### PR TITLE
Checks type for #2558

### LINKED ISSUE #2558: Trajectory.add_transformation doesn't fail cleanly
**Expected behavior**

A clear and concise description of what you want to do and what you think should happen. (Code to reproduce the behavior can be added below).

If I mess up adding a transformation, eg by passing a list of transformations in
```python
u.trajectory.add_transformations([transformations.unwrap(u.atoms)]) 
```

I get an error `TypeError: 'list' object is not callable`, which is fine, I didn't read the docs properly. But then if I try and do it again (this time correctly) I get: `ValueError: Can't add transformations again. Please create new Universe object`.

It looks like `add_transformations` isn't rolling back properly on an error, so instead you're stuck in some weird state where you can't get what you want without reloading.

### PR DESCRIPTION
**Pull request by Naimish240**
Fixes #2558 

Changes made in this Pull Request:
 checks type for list before passing into try-except block, and then checks for invalid type again inside the block for extra safety.

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/coordinates/base.py [hunk 1]
```diff
@@ -2045,8 +2045,28 @@ def add_transformations(self, *transformations):
...
             self.transformations = transformations
+        except TypeError:
+            # Check if nested list, if yes then 'remove' one layer of [] and apply transform
+            # Else, raises error asking for correct user input type
+            six.raise_from(
+                TypeError(
+                    "It seems as though you've passed an incompable object type!"
+                    "Please check the type being passed again"
+                )
+
```
Comment (RMeli) [inline package/MDAnalysis/coordinates/base.py]: There is a missing `)` here, this is why tests fail.

Comment (Naimish240) [reply package/MDAnalysis/coordinates/base.py]: I've added the ')' now. I'm extremely sorry for not seeing it earlier. I'll read up on the article to run tests locally before pushing code next time.
Thanks alot for your kind help and support!

**Comment by orbeckst** [comment]
See [User Guide: Testing](https://www.mdanalysis.org/UserGuide/testing.html) for some background.

File: package/MDAnalysis/coordinates/base.py [hunk 2]
```diff
@@ -2045,8 +2045,29 @@ def add_transformations(self, *transformations):
 
         """
 
+        # Checks if list is passed
+        if isinstance(transformations, list):
+            six.rais_from(
```
Comment (lilyminium) [inline package/MDAnalysis/coordinates/base.py]: [*args unpacking](https://realpython.com/python-kwargs-and-args/#using-the-python-args-variable-in-function-definitions) means that ``transformations`` will always be a tuple.

**Comment by orbeckst** [comment]
@shfrz do you intend to continue working on this PR?

**Comment by Naimish240** [comment]
Hey, sorry i couldn't get back earlier. I was caught up in Uni exams and forgot about this. I'm extremely sorry. I'll get back on this from Sunday, if it's fine with you.

**Comment by orbeckst** [comment]
I haven't heard anything here for the last 6 days so I am closing. If need be we can reopen – just ping us.

**Pull request closed without merging**